core/src/main/java/com/flipkart/gap/usl/core/store/dimension/kafka/ExternalKafkaPublisherDAOImpl.java

```diff
@@ -0,0 +1,55 @@
+package com.flipkart.gap.usl.core.store.dimension.kafka;
+
+import com.flipkart.gap.usl.core.config.ExternalKafkaConfig;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import com.google.inject.name.Named;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.common.record.CompressionType;
+
+import java.util.Properties;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingQueue;
+
+@Singleton
+@Slf4j
+public class ExternalKafkaPublisherDAOImpl extends KafkaPublisherDao{
+
+    @Inject
+    @Named("externalKafkaConfig")
+    private ExternalKafkaConfig externalKafkaConfig;
```

Comment:
Ideally we should not worry about external or internal kafka.. rather than creating separate config classes for external vs internalKafkaConfig we should create only kafkaConfig... and only one kafkaPublisherDao

Reply:
It will save this duplicated code as well